cognite/neat/workflows/steps_registry.py

```diff
@@ -0,0 +1,136 @@
+import importlib
+import inspect
+import os
+from pathlib import Path
+import sys
+from typing import Type
+
+from pydantic import BaseModel
+from cognite.neat.exceptions import InvalidWorkFlowError
+from cognite.neat.workflows.model import WorkflowConfigItem
+import cognite.neat.workflows.steps.lib
+import logging
+
+from cognite.neat.workflows.steps.step_model import DataContract, T_Output
+from cognite.neat.workflows.steps.step_model import Step
+
+
+class StepMetadata(BaseModel):
+    name: str
+    description: str = ""
+    category: str = "default"  # defines the category the step belongs to (e.g. "data", "model", "test")
+    scope: str = "global"  # defines the scope of the step (e.g. "global", local to a specific workflow)
+    input: list[str]
+    output: list[str]
+    configuration_templates: list[WorkflowConfigItem] = []
+
+
+class StepsRegistry:
+    def __init__(self, data_store_path: str | None = None):
+        self._step_classes: list[Step] = []
```

Comment:
```suggestion
        self._step_classes: dict[str, Step] = []
```
The way you use this one, you should use a dict instead.